org-roam.el

```diff
@@ -506,6 +522,53 @@ Bindings:
              (select-window org-roam-last-window))
     (find-file file)))
 
+(defun org-roam--unlinked-regexp (file-path)
+  (let* ((buffer-title (org-roam--get-title-or-slug file-path)))
+    (rx-to-string `(: word-boundary
+                      (or ,buffer-title
+                       ,(downcase buffer-title)
+                       ,(file-name-base file-path))
+                      word-boundary))))
+
+(defun org-roam--find-unlinked-references (file-path)
+  (let* ((match-regexp (org-roam--unlinked-regexp file-path))
+         (find-func (lambda (other-file)
+                      (when (not (string-equal other-file file-path))
+                        (with-temp-buffer
+                          (insert-file-contents other-file)
+                          (let ((matches)
+                                (unwanted-matches))
+                            (dolist (unwanted-regexp org-roam-unwanted-regexps)
+                              (while (re-search-forward unwanted-regexp nil t)
+                                (push (list :start (match-beginning 0) :end (match-end 0)) unwanted-matches))
+                              (goto-char (point-min)))
+                            (org-mode)
+                            (org-toggle-link-display)
+                            (while (re-search-forward match-regexp nil t)
+                              (let* ((p (match-beginning 0))
+                                     (e (match-end 0))
+                                     (context (buffer-substring (line-beginning-position) (line-end-position)))
+                                     (rep (format "[[file:%s][\\&]]" other-file))
+                                     (content (replace-regexp-in-string match-regexp rep context)))
+                                (org-toggle-link-display)
+                                (when (not (seq-filter (lambda (umatch)
+                                                    (and
+                                                     (> p (plist-get umatch :start))
+                                                     (< p (plist-get umatch :end))))
+                                                  unwanted-matches))
+                                  (push (list
+                                         :point p
+                                         :end e
+                                         :content content)
+                                        matches))
+                                (org-toggle-link-display)))
+                            (when (> (length matches) 0)
+                              (list
+                               :file-from other-file
+                               :buffer (buffer-string)
+                               :matches matches))))))))
+    (delete nil (mapcar find-func (org-roam--find-all-files)))))
```

Comment:
This is slow. Can we fork this out to some shell program (rg, grep etc.)? Asynchronously perform the search, and update the buffer.

Reply:
That is one solution. However, if we did that, we would miss out on some benefits like knowing the precise point in the buffer for the link. Would it not be better to do it asynchronously in emacs itself (e.g using async)?

Reply:
> we would miss out on some benefits like knowing the precise point in the buffer for the link.

Most search programs return row and column number, I think that could be used to go to point as well.

> Would it not be better to do it asynchronously in emacs itself (e.g using async)?

Yeah, this is also a good solution.